python/cudf/cudf/tests/test_indexing.py

```diff
@@ -1741,3 +1742,319 @@ def test_boolean_mask_columns_iloc_series():
+
+
+@pytest.mark.xfail(reason="https://github.com/rapidsai/cudf/issues/13267")
+@pytest.mark.parametrize(
+    "indexer", [(..., 0), (0, ...)], ids=["row_ellipsis", "column_ellipsis"]
+)
+def test_iloc_13267(indexer):
+    df = pd.DataFrame(np.arange(4).reshape(2, 2))
+    cdf = cudf.from_pandas(df)
+
+    expect = df.iloc[indexer]
+    actual = cdf.iloc[indexer]
+    assert_eq(expect, actual)
+
+
+@pytest.mark.parametrize(
+    "indexer",
+    [
+        0,
+        (slice(None), 0),
+        pytest.param(
+            ([0, 2], 1),
+            marks=pytest.mark.xfail(
+                reason="https://github.com/rapidsai/cudf/issues/13515"
+            ),
+        ),
+        (slice(None), slice(None)),
+        (slice(None), [1, 0]),
+        (0, 0),
+        (1, [1, 0]),
+        pytest.param(
+            ([1, 0], 0),
+            marks=pytest.mark.xfail(
+                reason="https://github.com/rapidsai/cudf/issues/13515"
+            ),
+        ),
+        pytest.param(
+            ([1, 2], [0, 1]),
+            marks=pytest.mark.xfail(
+                reason="https://github.com/rapidsai/cudf/issues/13515"
+            ),
+        ),
+    ],
+)
+def test_iloc_multiindex_13515(indexer):
+    df = pd.DataFrame(
+        {"a": [1, 1, 3], "b": [2, 3, 4], "c": [1, 6, 7], "d": [1, 8, 9]}
+    ).set_index(["a", "b"])
+    cdf = cudf.from_pandas(df)
+
+    expect = df.iloc[indexer]
+    actual = cdf.iloc[indexer]
+    assert_eq(expect, actual)
+
+
+@pytest.mark.xfail(reason="https://github.com/rapidsai/cudf/issues/12833")
+def test_loc_12833_unordered():
+    df = pd.DataFrame({"a": [1, 2, 3]}, index=[7, 0, 4])
+    cdf = cudf.from_pandas(df)
+
+    # Check that pandas don't change their mind
+    with pytest.raises(KeyError):
+        df.loc[1:5]
+
+    with pytest.raises(KeyError):
+        cdf.loc[1:5]
+
+
+@pytest.mark.xfail(reason="https://github.com/rapidsai/cudf/issues/13379")
+@pytest.mark.parametrize("index", [range(5), list(range(5))])
+def test_loc_13379_keyerror_missing(index):
+    df = pd.DataFrame({"a": index}, index=index)
+    cdf = cudf.from_pandas(df)
+    # Check that pandas don't change their mind
+    with pytest.raises(KeyError):
+        df.loc[[0, 5]]
+
+    with pytest.raises(KeyError):
+        cdf.loc[[0, 5]]
+
+
+class TestLocIndexWithOrder:
+    @pytest.fixture(params=["increasing", "decreasing", "neither"])
+    def order(self, request):
+        return request.param
+
+    @pytest.fixture(params=[-1, 1], ids=["reverse", "forward"])
+    def take_order(self, request):
+        return request.param
+
+    @pytest.fixture(params=["float", "int", "string"])
+    def dtype(self, request):
+        return request.param
+
+    @pytest.fixture
+    def index(self, order, dtype):
+        if dtype == "string":
+            index = ["a", "h", "f", "z"]
+        elif dtype == "int":
+            index = [-1, 10, 7, 14]
+        elif dtype == "float":
+            index = [-1.5, 7.10, 2.4, 11.2]
```

Comment:
It seems like these lists should just _be_ the parameters, rather than having a parameter `dtype`. It doesn't seem that `dtype` is used otherwise.

Reply:
I use the dtype to conditionally apply a marker in the actual test:

```
        if not (order == "increasing" and dtype in {"int", "float"}):
            request.applymarker(
                pytest.mark.xfail(
                    reason="https://github.com/rapidsai/cudf/issues/12833"
                )
            )
```

So it's somewhat easier to just know that rather than pulling the dtype out of the index on the dataframe.